src/poetry_plugin_freeze/app.py

```diff
@@ -22,10 +23,18 @@
 from poetry.core.version.markers import union as marker_union
 from poetry.factory import Factory
 from poetry.plugins.application_plugin import ApplicationPlugin
+from poetry.core.packages.utils.utils import create_nested_marker
+from poetry.core.version.markers import parse_marker
 
 from poetry_plugin_export.walker import get_project_dependency_packages, walk_dependencies
 
 
+is_legacy_poetry = parse_semver(poetry.core.__version__) < parse_semver("2")
+
+if is_legacy_poetry:
+    from poetry.core.pyproject.exceptions import PyProjectException
```

Comment:
This came from an [exception rename](https://github.com/python-poetry/poetry/pull/9729/files#diff-c7d904b3993aceb26aa850119fdae2cfb308fdfbddefba3e8fbd0d974371c872R94) on Poetry's end. We may be better off trying to import both forms, rather than trying to selectively import based on version:

```suggestion
try:
    from poetry.core.pyproject.exceptions import PyProjectError
except ImportError:
    from poetry.core.pyproject.exceptions import PyProjectException as PyProjectError
```

...which would let us catch `PyProjectError` directly below.

Reply:
This is actually a good point, and i also applied it.

But there is one point, you need to know: in the situation, that we're going to capture, poetry 2.x is raising a RuntimeError instead of the new PyProjectError. I dont know why, but i stumbled above this pitfall, since i first also assumed, that it is like you described.

Reply:
Good point, I think we need to handle those 2 cases separately. One is the `PyProjectError` (formerly `PyProjectException`) which can come up in some cases (like https://github.com/cloud-custodian/poetry-plugin-freeze/issues/14).

Other issues can raise a `RuntimeError` like you mentioned. In either case we should skip freezing that wheel and move on.